src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.controller.js

```diff
@@ -0,0 +1,57 @@
+(function () {
+    "use strict";
+
+    function PartialViewMacrosCreateController($scope, codefileResource, $location, navigationService) {
+
+        var vm = this;
+        var node = $scope.dialogOptions.currentNode;
+
+        vm.snippets = [];
+        vm.showSnippets = false;
+        vm.creatingFolder = false;
+
+        vm.createPartialViewMacro = createPartialViewMacro;
+        vm.showCreateFolder = showCreateFolder;
+        vm.createFolder = createFolder;
+        vm.showCreateFromSnippet = showCreateFromSnippet;
+
+        function onInit() {
+            codefileResource.getSnippets('partialViewMacros')
+                .then(function(snippets) {
+                    vm.snippets = snippets;
+                });
+        }
+
+        function createPartialViewMacro(selectedSnippet) {
+
+            var snippet = null;
+
+            if(selectedSnippet && selectedSnippet.fileName) {
+                snippet = selectedSnippet.fileName;
+            }
+
+            $location.search('create', null);
+            $location.search('snippet', null);
+            $location.path("/developer/partialviewmacros/edit/" + node.id).search("create", "true").search("snippet", snippet);
+            navigationService.hideMenu();
+
+        }
+
+        function showCreateFolder() {
+            vm.creatingFolder = true;
+        }
+
+        function createFolder() {
+            alert("create folder");
```

Comment:
I know this is coming in a future issue @madsrasmussen & @emilwangaa - but be good not to commit alert/console.log debugging stuff.